mysql-test/main/subselect_mat.test

```diff
@@ -296,3 +296,50 @@ UNION ALL
 ;
 
 drop table t1, t2;
+
+--echo #
+--echo # MDEV-34665: Simplify IN predicate processing for NULL-aware
+--echo #             materialization involving only one column
+
+set @save_optimizer_switch=@@optimizer_switch;
+set @@optimizer_switch = "materialization=on,in_to_exists=off,semijoin=off";
+
+create table t1 (a int);
+create table t2 (b int);
+insert into t1 values (null), (1), (2), (3);
+
+--echo # Query against empty t2
+--sorted_result
+select * from t1 where a not in (select b from t2);
+--sorted_result
+select a from t1 join t2 on a not in (select b from t2);
+--sorted_result
+select a from t1 left join t2 on a not in (select b from t2);
+
```

Comment:
As discussed on Slack: the 3 variants are excessive. 
It's better to put the subquery into the select list:
```
select a,  a in (select ....) as SUBQ ,,,
```
This puts the subquery into a context where the subquery's arguments and return value can be easily seen.   (and the subquery is not a "top-level item" so optimizer doesn't take shortcuts).   Using a join outside the subquery is redundant.